src/components/organisms/account-data/account-data.tsx

```diff
@@ -1,80 +1,94 @@
+import styles from './account-data.module.scss';
+
+const schema = getValidationSchema<AccountFormData>(
+  'email',
+  'phone',
+  'password'
+);
+
+const initialValues: AccountFormData = {
+  email: '',
+  phone: '',
+  password: '',
+};
 
 /**
  * Компонент-виджет с редактируемой формой данных аккаунта.
  * */
 
 export const AccountData: FC = () => {
-  const { handleChange, isValid, errors, values, setValues, setIsValid } =
-    useFormWithValidation<AccountFormData>();
-
   const accountData = useAppSelector<AccountFormData>(selectProfileAccount);
   const dispatch = useAppDispatch();
 
+  const handleSubmit = async (
+    values: AccountFormData,
+    { validateForm }: FormikHelpers<AccountFormData>
+  ) => {
+    await validateForm(values);
+    void dispatch(setAccountData(values));
+  };
+
+  const formik = useFormik<AccountFormData>({
+    initialValues,
+    validationSchema: schema,
+    onSubmit: handleSubmit,
+  });
+
+  const fieldsToCompare = ['phone'];
+
+  const areSameValues = compareObjectFields(
+    accountData,
+    formik.values,
+    fieldsToCompare
+  );
+
   useEffect(() => {
-    setValues(accountData);
+    void formik.setValues(accountData);
   }, [accountData]);
 
-  const handleFormSubmit = (evt: FormEvent<HTMLFormElement>) => {
-    evt.preventDefault();
-    dispatch(setAccountData(values));
-    setIsValid(false);
-  };
-
   return (
     <Card className={styles.accountData}>
       <Heading level={3} size="l" weight="bold" text="Аккаунт" />
 
       <form
         name="accountDataForm"
-        onSubmit={handleFormSubmit}
+        onSubmit={formik.handleSubmit}
         className={styles.form}
+        noValidate
       >
         <Input
           labelName="Номер телефона"
           type="tel"
-          name="phoneNumber"
-          value={values.phoneNumber || ''}
-          onChange={handleChange}
+          name="phone"
+          value={formik.values.phone}
+          onChange={formik.handleChange}
           placeholder="Номер телефона начиная с +7"
-          disabled
+          isWithIcon
         />
         <Input
           labelName="Email"
           type="email"
           name="email"
-          value={values.email || ''}
-          onChange={handleChange}
-          isValid={!errors.email}
-          error={errors.email}
-          isWithIcon
+          value={formik.values.email}
+          onChange={formik.handleChange}
           placeholder="Ваш email"
-        />
-        <Input
-          labelName="Пароль"
-          type="password"
-          name="password"
-          value={values.password || ''}
-          onChange={handleChange}
-          placeholder="Ваш пароль"
-          isWithIcon
-          minLength={8}
-          isValid={!errors.password}
-          error={errors.password}
+          disabled
         />
         <Button
           type="submit"
-          disabled={!isValid}
+          disabled={areSameValues && (formik.isSubmitting || !formik.isValid)}
```

Comment:
```suggestion
          disabled={areSameValues || formik.isSubmitting}
```
С `!formik.isValid` будут проблемы, лучше без него